### PR TITLE
Api 4766- Lockdown publish endpoints V2

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
- Change auth failure to a 401 (Unauthorized)
- Add auth to delete integration endpoint

### CODE REVIEW

File: app/uk/gov/hmrc/integrationcatalogueadminfrontend/config/JsonOrHtmlErrorHandler.scala [hunk 1]
```diff
@@ -40,7 +40,10 @@ class JsonOrHtmlErrorHandler @Inject()(environment: Environment, htmlErrorHandle
     else {
       val errorMessage = statusCode match {
         case NOT_FOUND  => s"Path or Http method may be wrong. $message"
-        case _ => message
+        case _ => {
+          println(s"message: $message ($statusCode)")
+          message
+        }
```
Comment (adampridmore) [inline app/uk/gov/hmrc/integrationcatalogueadminfrontend/config/JsonOrHtmlErrorHandler.scala:46]: We should remove this! I think I added it when I was 'debugging'

File: app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/IntegrationController.scala [hunk 2]
```diff
@@ -62,11 +66,15 @@ class IntegrationController @Inject()(appConfig: AppConfig,
       }
  }
 
-  def deleteByIntegrationId(integrationId: IntegrationId) : Action[AnyContent] =
-    Action.async { implicit request =>
-    integrationService.deleteByIntegrationId(integrationId).map {
-      case true => NoContent
-      case false => NotFound(Json.toJson(ErrorResponse(List(ErrorResponseMessage(s"deleteByIntegrationId: The requested resource could not be found.")))))
+  def deleteByIntegrationId(integrationId: IntegrationId) : Action[AnyContent] = {
+      println(s"deleteByIntegrationId.integrationId: $integrationId")
```
Comment (adampridmore) [inline app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/IntegrationController.scala:70]: We should remove this println! I think I added it when I was 'debugging'

File: app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/binders/package.scala [hunk 3]
```diff
@@ -25,8 +25,8 @@ import play.api.mvc.QueryStringBindable
 
 package object binders {
 
-  private def integrationIdFromString(text: String): Either[String, IntegrationId] = {
-    Try(UUID.fromString(text))
+  def integrationIdFromString(text: String): Either[String, IntegrationId] = {
+   Try(UUID.fromString(text))
```
Comment (adampridmore) [inline app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/binders/package.scala:29]: This should be private again.

File: conf/services.routes [hunk 4]
```diff
@@ -12,3 +12,4 @@ GET            /integrations                                uk.gov.hmrc.integrat
 GET            /integrations/:id                            uk.gov.hmrc.integrationcatalogueadminfrontend.controllers.IntegrationController.findByIntegrationId(id: uk.gov.hmrc.integrationcatalogue.models.common.IntegrationId)
 + nocsrf
 DELETE         /integrations/:id                            uk.gov.hmrc.integrationcatalogueadminfrontend.controllers.IntegrationController.deleteByIntegrationId(id: uk.gov.hmrc.integrationcatalogue.models.common.IntegrationId)
+# DELETE         /integrations/:id                            uk.gov.hmrc.integrationcatalogueadminfrontend.controllers.IntegrationController.deleteByIntegrationId(id: String)
```
Comment (adampridmore) [inline conf/services.routes:15]: Delete me!

File: it/controllers/IntegrationControllerISpec.scala [hunk 5]
```diff
@@ -62,19 +63,23 @@ class IntegrationControllerISpec extends ServerBaseISpec with BeforeAndAfterEach
...
+    }
   }
 
   "IntegrationController" when {
 
     "DELETE [some unknown path]" should {
       "return blah" in new Setup {
          val response: Future[Result] = route(app, invalidPathRequest).get
-         status(response) mustBe NOT_FOUND
+        //  status(response) mustBe NOT_FOUND
```
Comment (adampridmore) [inline it/controllers/IntegrationControllerISpec.scala:82]: Un-comment me.

File: it/controllers/IntegrationControllerISpec.scala [hunk 6]
```diff
@@ -213,7 +218,13 @@ class IntegrationControllerISpec extends ServerBaseISpec with BeforeAndAfterEach
...
      "respond with 400 when non uuid id provided" in new Setup {
 
-        val response: Future[Result] = route(app, validDeleteIntegrationRequest("invalidId")).get
+        // Invalid with auth header
+        //val request = validDeleteIntegrationRequest("invalidId")
+        
+        // TODO: Invalid with no auth header
+        val request = FakeRequest(Helpers.DELETE, s"/integration-catalogue-admin-frontend/services/integrations/invalidId")
+
+        val response: Future[Result] = route(app, request).get
```
Comment (adampridmore) [inline it/controllers/IntegrationControllerISpec.scala:227]: Comment out bits should be removed.

File: it/controllers/IntegrationControllerISpec.scala (hunk 5 above)
Comment (mattclark-zerogravit) [reply it/controllers/IntegrationControllerISpec.scala]: done

File: conf/services.routes (hunk 4 above)
Comment (mattclark-zerogravit) [reply conf/services.routes]: done

File: app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/binders/package.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/integrationcatalogueadminfrontend/controllers/binders/package.scala]: done

**Pull request merged**